File: src/main/java/inno/jago/lexer/Pos.java
package inno.jago.lexer;

import inno.jago.ast.entity.Entity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class Pos extends Entity {

    private long line;

    private int symbol;

    void report() {
        System.out.print(line + ":" + symbol);
    }
}

File: src/main/kotlin/inno/jago/lexer/Pos.kt
package inno.jago.lexer

import org.antlr.v4.runtime.Token

data class Pos (
    val line: Int,
    val startIndex: Int,
    val stopIndex: Int
) {
    constructor(token: Token) : this(token.line, token.startIndex, token.stopIndex)
    constructor(tokenStart: Token, tokenStop: Token) :
            this(tokenStart.line, tokenStart.charPositionInLine, tokenStop.charPositionInLine)

    override fun toString(): String =
        "line $line:$startIndex-$stopIndex"
}
